keras/src/callbacks/orbax_checkpoint.py

```diff
@@ -0,0 +1,294 @@
+import warnings
+
+import numpy as np
+
+from keras.src import backend
+from keras.src import tree
+from keras.src.api_export import keras_export
+from keras.src.callbacks.monitor_callback import (
+    MonitorCallback,  # For metric monitoring logic
+)
+from keras.src.utils.io_utils import print_msg
+from keras.src.utils.module_utils import ocp
+
+# Context and AsyncOptions are accessed through the lazy-loaded ocp module
+
+
+def _get_state_tree(model):
+    """Get the complete model state as a nested tree structure."""
+    # For JAX backend, preserve native arrays if JAX >= 0.7.0
+    # to avoid unnecessary conversions. Otherwise convert to numpy.
+    did_numpy_conversion = False
+    if backend.backend() == "jax":
+        import jax
+        from packaging import version
+
+        # Check JAX version directly (JAX 0.7.0+ supports better array handling)
+        if version.parse(jax.__version__) >= version.parse("0.7.0"):
+            state_tree = model.get_state_tree()
+        else:
+            # Fallback to numpy conversion for older JAX versions
+            state_tree = model.get_state_tree(value_format="numpy_array")
+            did_numpy_conversion = True
```

Comment:
What is the issue with JAX < 0.7.0 ?

Orbax checkpointing was available way before 0.7.0 so I'm really surprised.

Also, we have issues with our CI that prevents us from upgrading JAX, so we won't be testing the "real" use case, which is not great.